Add JavaScript tests for the `PhoneNumberCard` component

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This is a part of 📌 [Add tests](https://github.com/woocommerce/google-listings-and-ads/issues/1993#fe-tests) in #1993.

Before adding tests for the `VerifyPhoneNumberContent` component, I would like to add tests for the `PhoneNumberCard` component first.

### Detailed test instructions:

1. Check if the newly added tests fit with the test purpose.
2. Check if all JS tests can pass.

### Changelog entry

eason9487 committed:
Add JS tests for the `PhoneNumberCard` component.

puntope commented on js/src/components/contact-information/phone-number-card/phone-number-card.test.js:
```diff
@@ -0,0 +1,262 @@
+	} );
+
+	it( 'When `initEditing` is not specified, should render in display mode after loading a verified phone number', () => {
+		mockLoaded( false );
+		const { rerender } = render(
+			<PhoneNumberCard phoneNumber={ phoneNumber } />
+		);
```
I guess this can be omitted.

eason9487 replied:
This is to simulate the state change of "after loading a verified phone number".

puntope commented on js/src/components/contact-information/phone-number-card/phone-number-card.test.js:
```diff
@@ -0,0 +1,262 @@
+		expect( button ).not.toBeInTheDocument();
+	} );
+
+	it( 'When `initEditing` is not specified, should render in display mode after loading a verified phone number', () => {
```
"display mode"  doesn't explain much. I guess it would be more accurate to say 

should render the edit button after loading a verified phone number

eason9487 replied:
The mode for entering verification code also has an edit button. Changed to use "non-editing mode" in https://github.com/woocommerce/google-listings-and-ads/pull/2027/commits/ca7ce02bbb5aa9e407dfe26ac47ecff3be8344a9.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/c9a97fd5-0b03-4f72-8b72-dd596bf620c5)

puntope commented on js/src/components/contact-information/phone-number-card/phone-number-card.test.js:
```diff
@@ -0,0 +1,262 @@
+	} );
+
+	it( 'When `initEditing` is not specified, should render in editing mode after loading an unverified phone number', () => {
+		mockLoaded( false );
+		const { rerender } = render(
+			<PhoneNumberCard phoneNumber={ phoneNumber } />
+		);
```
I guess same as before, this seems not necessary

eason9487 replied:
This is to simulate the state change of "after loading an unverified phone number".

puntope reviewed (approved):
✅  LGTM 

Tests pass and seem to cover all the scenarios.

eason9487 committed:
Change "display mode" to "non-editing mode" in the description of `PhoneNumberCard` tests.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2027#discussion_r1277807856
